android/src/main/java/io/customer/reactnative/sdk/CustomerIOReactNativeModule.kt

```diff
@@ -1,105 +1,177 @@
+        return isInstanceInvalid
     }
 
     @JvmOverloads
     @ReactMethod
-    fun initialize(siteId: String, apiKey: String, region: String? = null) {
-        if (isInitialized()) return
-
-        customerIO = CustomerIO.Builder(
-            siteId = siteId,
-            apiKey = apiKey,
-            region = region.toRegion() ?: Region.US,
-            appContext = application,
-        )
-            .setLogLevel(CioLogLevel.DEBUG)
-            .autoTrackDeviceAttributes(true)
-            .build()
+    fun initialize(
+        environment: ReadableMap,
+        configuration: ReadableMap? = null,
+        sdkVersion: String? = null,
+    ) {
+        if (isInstanceValid()) {
+            logger.info("Customer.io instance already initialized")
+            return
+        }
 
+        val env = environment.toMap()
+        val config = configuration?.toMap()
+
+        try {
+            val siteId = env.getString(Keys.Environment.SITE_ID)
+            val apiKey = env.getString(Keys.Environment.API_KEY)
+            val region = env.getProperty<String>(
+                Keys.Environment.REGION
+            )?.takeIfNotBlank().toRegion()
+            val organizationId = env.getProperty<String>(
+                Keys.Environment.ORGANIZATION_ID
+            )?.takeIfNotBlank()
+
+            customerIO = CustomerIO.Builder(
+                siteId = siteId,
+                apiKey = apiKey,
+                region = region,
+                appContext = reactApplicationContext.applicationContext as Application,
+            ).apply {
+                // TODO: Update when sdk version is released
+                setClient(Client.ReactNative)
+                setupConfig(config)
+                addCustomerIOModule(module = configureModuleMessagingPushFCM(config))
+                if (!organizationId.isNullOrBlank()) {
+                    addCustomerIOModule(module = configureModuleMessagingInApp(organizationId))
+                }
+            }.build()
+        } catch (ex: IllegalArgumentException) {
+            logger.error(ex.message ?: "$MODULE_NAME -> initialize -> IllegalArgumentException")
+        }
+        CustomerIOReactNativeInstance.onSDKInitialized()
+        val fcmToken = CustomerIOReactNativeInstance.cachedFCMToken
+        if (!fcmToken.isNullOrBlank() && CustomerIOReactNativeInstance.setFCMTokenRegistered()) {
+            CustomerIOFirebaseMessagingService.onNewToken(fcmToken)
+        }
     }
 
-    @ReactMethod
-    fun config(attributes: ReadableMap?) {
-        if (isInitialized()) {
-            logMessage("Customer.io instance already initialized")
-            return
+    private fun CustomerIO.Builder.setupConfig(config: Map<String, Any>?): CustomerIO.Builder {
+        if (config == null) return this
+
+        val logLevel = config.getProperty<Double>(Keys.Config.LOG_LEVEL).toCIOLogLevel()
+        CustomerIOReactNativeInstance.setLogLevel(logLevel = logLevel)
+        setLogLevel(level = logLevel)
```

Comment:
This is confusing me, you are setting `logLevel` using `CustomerIOReactNativeInstance` and somewhere in this class, how are these two different ?

Reply:
The first `logLevel` is default value based on build type as we don't have user preference at that point. As soon as we get the user preference value, we update if for both `CustomerIOReactNativeInstance` (ReactNative SDK) and `setLogLevel` (Android SDK)

Reply:
By `CustomerIOReactNativeInstance` you mean the android side of react package for which you have created logger class to show logs based on preference? Please correct me if I am wrong.

Reply:
Yeah exactly, you are correct. `CustomerIOReactNativeInstance` is just holding stuff for Android part of react package to make communication with our Android SDK easier.